lib/que/adapters/active_record_with_lock.rb

```diff
@@ -35,13 +47,22 @@ def execute(command, params = [])
       def lock_job_with_lock_database(queue, cursor)
         result = []
         loop do
-          result = Que.execute(:find_job_to_lock, [queue, cursor])
+          result = Que.transaction do
+            observe(nil, FindJobSecondsTotal, queue: queue) do
+              result = Que.execute(:find_job_to_lock, [queue, cursor])
+            end
 
-          break if result.empty?
+            return result if result.empty?
 
-          cursor = result.first["job_id"]
-          break if pg_try_advisory_lock?(cursor)
+            cursor = result.first["job_id"]
+            job_locked = pg_try_advisory_lock?(cursor)
```

Comment:
Do we want to include the SQL query here in the metric for the time it takes to find a job ?

Reply:
Nit + non blocking for this PR: also, if you unify the metric labels you might be able to use `Hesiod.register_duration_counters` and `Hesiod.observe` as convenience wrappers

Reply:
time to find a job is in line 51. Locking metric shouldn't take much time. Not required to track as of now. 
There is already a metric within Locker to track this